atom/index.d.ts

```diff
@@ -99,6 +99,64 @@ export interface WritableAtom<Value = any> extends ReadableAtom<Value> {
 
 export type Atom<Value = any> = ReadableAtom<Value> | WritableAtom<Value>
 
+export interface Autosubscribe<Value = any> {
+  /**
+   * @return The computed store's value
+   */
+  (): Value
```

Comment:
I don’t see those change in `atom/index.js` and tests use `use()` method instead

Reply:
There are some conversations either incomplete or not yet had with the details of this PR. The PR has evolved from a the scope of just autosubscribing to including async & supporting some important edge cases. I hope to bring more clarity to these discussions.

> What is the different with .get()?

`.get` does not autosubscribe & keeps the same behavior that is currently in nanostores. 

> I don’t see those change in atom/index.js and tests use use() method instead.

> Can we use it only for autosubscriber()’s results?

`autosubscriber` should be renamed `autosubscribe`, as it is an instance of `Autosubscribe`.

Some context on the choise to use `use`...I'm looking for a short name to reduce api noise introduced by autosubscribe within the computed callback. Calling `autosubscribe($my-store)` is verbose, particularly when used in close proximity to other calls to `autosubscribe`. So I previously shortened `autosubscribe` to `asub`.

I started using `use()` in the tests in response to https://github.com/nanostores/nanostores/issues/188#issuecomment-1611611341. I'm hoping for us to agree on a convention. Note that `use` is also an object with member functions, such as `()`, `($atom: ReadableAtom)`, `.stale()`, `.set(val:Value)`, `.save(val:Value)`, `.on()`, `.off()`, etc.

The most terse api with the least noise is to make `atom` a function which autosubscribes the `atom` to the calling `computed`. I removed that behavior from the [PR branch](https://github.com/btakita/nanostores/tree/issues/188) in response to https://github.com/nanostores/nanostores/issues/188#issuecomment-1611701108. See [my reply](https://github.com/nanostores/nanostores/issues/188#issuecomment-1612032245) for more context.

The branch where atom is a function can be found at https://github.com/btakita/nanostores/tree/issues/188-atom-fn. I explained the change in [this comment](https://github.com/nanostores/nanostores/issues/188#issuecomment-1612032245). I can have this PR use the [issues/188-atom-fn](https://github.com/btakita/nanostores/tree/issues/188-atom-fn) branch if you agree with me on this point. If it were up to me, I would have this PR use the [issues/188-atom-fn](https://github.com/btakita/nanostores/tree/issues/188-atom-fn) branch, however I also do not want this to cause confusion by having both `$computed()` & `use($computed)` effectively be aliases to each other. I personally do not think it's confusing, but others may think it's confusing.

Reply:
> I don’t see those change in atom/index.js and tests use use() method instead.
> Can we use it only for autosubscriber()’s results?

I also want to clarify the use case for the `Autosubscribe` `()` function. Inside the `computed` callback, there is no reliable way to have a reference to `$computed`. `()`, along with `.on()` & `.off()` (see the review topic below), support the use case of behaviorally accessing `$computed` within the computed callback.

Reply:
> I personally think the terser $computed() is beneficial to reduce api noise...so I will use the atom function defined either in nanostores or in another library that extends nanostores

If we choose to not include the `$atom()` function in nanostores, perhaps a library which imports nanostores named "autostores" or "auto-nanostores" could be created to add the `$atom()` function behavior. This would allow the programmer to explicitly opt-into this behavior, bringing more awareness to how the implicit `autosubscribe` to `$atom()` behaves with it's benefits & inconsistencies.

Reply:
I renamed `autosubscriber` to `autosubscribe`.

Reply:
The fact that using `.get()` will not make auto-subscriptions look strange for me. Some developer could forget it and use `get()` by a mistake and got an error which is very hard to debug.

Reply:
> Some developer could forget it and use get() by a mistake and got an error which is very hard to debug.

That is definitely something to consider with the ergonomics of using autosubsciptions.

I found that having a way to get the current state of a store without autosubscribing is useful in some cases, which will be more common with async computeds. This could be to avoid circular subscriptions causing a loop or referencing a store where a subscription is not wanted, such as a logger:

```ts
let $logger = atom()
let $userId = atom()
let $user = computed(use => {
  let userId = use($userId)
  if (userId == null) {
    info('No userId')
    return null
  }
  info('fetching userId: ', userId)
  return fetch(`https://my-api/users/${userId}`).then(response => response.json())
  function info(...argv) {
    let logger = $logger.get()
    if (logger) logger.info(...argv)
  }
})
```

Having the `$atom()` function would be helpful for these issues. Imagine `$atom()` being the primary way to get the value of the atom & autosubscribe, reserving `.get()` for special cases inside of a `computed` callback where an autosubscription is not wanted. That way, the normal (> ~98%) case is to simply call `$atom()` leaving `.get()` to be used in a small number of cases with the specific intent of getting the value without an autosubscription. In some of my codebases, I converted to using `$atom()` & noticed that `$atom()` is called > 98% of the time, with a few calls to `.get()` inside of some computed calbacks.

This leads to the other inconsistencies with using the `$atom()` function. The inconsistencies would only need to occur in an async `computed` callback when the synchronous task completes, which will be less often.

These inconsistencies could be documented with `computed`. So the idiom would be to use `$atom()` most of the time, `.get()` within a computed where autosubscribe is not wanted, & `use($atom)` or `$atom(use)` when the we want to autosubscribe after the synchronous task completes.

Reply:
With that being said, I converted this pr branch `issues/188` to have the `$atom()` function. I created another branch where atom is not a function (the previous code for this PR) at https://github.com/btakita/nanostores/tree/issues/188-no-atom-fn.

Reply:
Another reason to keep the existing behavior of `.get` is preventing errors when upgrading to the version that contains this PR.

Reply:
Personally I agree about `.get()` not automatically subscribing. This makes it consistent with UI library bindings, where calling `.get()` inside a React component also won't make the component subscribe to the store, to subscribe `useStore` must be used instead of calling `.get()` directly.

Reply:
> This makes it consistent with UI library bindings, where calling .get() inside a React component also won't make the component subscribe to the store, to subscribe useStore must be used instead of calling .get() directly

Actually, autosubscribe would not occur if you use `$computed()` or `$computed.get`. Autosubscribe only occurs within the the synchronous task of a `computed` callback. However, I still think `.get` should keep it's existing behavior & not autosubscribe, so existing apps don't break.